From 6f99fc0866853f579496376a52bc7ce17451e510 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sun, 22 Apr 2018 17:50:18 +0200
Subject: [PATCH 10/12] PhoneWindowManager: Don't timeout when taking a partial
 screenshot

* A partial screenshot shouldn't timeout when the user might still be
  adjusting the rectangle

BUGBASH-1500

Change-Id: Ifb06173dc712667db3bfb3c6f945b9bddb2a85fa
---
 .../java/com/android/server/policy/PhoneWindowManager.java    | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 949d7306b97..6b14632d5b7 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -6474,7 +6474,9 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                     Context.BIND_AUTO_CREATE | Context.BIND_FOREGROUND_SERVICE_WHILE_AWAKE,
                     UserHandle.CURRENT)) {
                 mScreenshotConnection = conn;
-                mHandler.postDelayed(mScreenshotTimeout, 10000);
+                if (screenshotType != WindowManager.TAKE_SCREENSHOT_SELECTED_REGION) {
+                    mHandler.postDelayed(mScreenshotTimeout, 10000);
+                }
             }
         }
     }
-- 
2.17.0

